evalml/pipelines/components/estimators/regressors/lightgbm_regressor.py

```diff
@@ -0,0 +1,96 @@
+import copy
+
+import pandas as pd
+from sklearn.preprocessing import OrdinalEncoder
+from skopt.space import Integer, Real
+
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components.estimators import Estimator
+from evalml.problem_types import ProblemTypes
+from evalml.utils import SEED_BOUNDS, get_random_seed, import_or_raise
+from evalml.utils.gen_utils import (
+    _convert_to_woodwork_structure,
+    _convert_woodwork_types_wrapper,
+    _rename_column_names_to_numeric
+)
+
+
+class LightGBMRegressor(Estimator):
+    """LightGBM Regressor"""
+    name = "LightGBM Regressor"
+    hyperparameter_ranges = {
+        "learning_rate": Real(0.000001, 1),
+        "boosting_type": ["gbdt", "dart", "goss", "rf"],
+        "n_estimators": Integer(10, 100),
+        "max_depth": Integer(0, 10),
+        "num_leaves": Integer(2, 100),
+        "min_child_samples": Integer(1, 100),
+        "bagging_fraction": Real(0.000001, 1),
```

Comment:
Is 0 an invalid setting for `bagging_fraction`?

Reply:
Yep, LightGBM requires it to be strictly larger than 0 when provided